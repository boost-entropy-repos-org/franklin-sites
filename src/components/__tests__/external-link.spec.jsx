import React from 'react';
import renderer from 'react-test-renderer';

import ExternalLink from '../external-link';

describe('ExternalLink component', () => {
  test('should render external link', () => {
    const component = renderer
      .create(
        <ExternalLink url="#">
          <span>Link text</span>
        </ExternalLink>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render external link without target: _blank attribute', () => {
    const component = renderer
      .create(
        <ExternalLink url="#" newTab={false}>
          <span>Link text</span>
        </ExternalLink>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
