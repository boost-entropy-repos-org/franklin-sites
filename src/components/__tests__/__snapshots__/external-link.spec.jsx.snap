// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink component should render external link 1`] = `
<a
  className="external-link"
  href="#"
  rel="noopener"
  target="_blank"
>
  <span>
    Link text
  </span>
  <test-file-stub
    width={12.5}
  />
</a>
`;

exports[`ExternalLink component should render external link without target: _blank attribute 1`] = `
<a
  className="external-link"
  href="#"
  rel="noopener"
>
  <span>
    Link text
  </span>
  <test-file-stub
    width={12.5}
  />
</a>
`;
